bigtop-packages/src/charm/spark/layer-spark/reactive/spark.py

```diff
@@ -236,7 +235,8 @@ def reconfigure_spark():
     # Almost all config changes should trigger a reinstall... except when
     # changing the bigtop repo version. Repo version changes require the user
     # to run an action, so we skip the reinstall in that case.
-    if not is_state('config.changed.bigtop_version'):
+    if not is_state('config.changed.bigtop_version') \
+            and not is_state('config.changed.spark_enable_thriftserver'):
```

Comment:
I'm a bit unsettled with this.  If someone changed multiple config options with something like `juju config spark driver_memory=2048M spark_enable_thriftserver=true`, we'd miss the driver_mem change.

What if we adjusted the `reconfigure_spark` function to be:
```
@when('spark.started', 'config.changed')
@when_not('config.changed.bigtop_version', 'config.changed.spark_enable_thriftserver')
 def reconfigure_spark():
...
```

We could then remove the check for `config.changed.bigtop_version` in the `reconfigure_spark` handler (it gets handled anyway with the `check_repo_version` function a few lines down). Then we could adjust your `start_thrift` function to be:
```
@when('spark.started', 'config.changed.spark_enable_thriftserver')
def start_thrift():
    enable_thrift = hookenv.config()['spark_enable_thriftserver']
    if enable_thrift:
        check_call(['/usr/lib/spark/sbin/start-thriftserver.sh'])
    else:
        check_call(['/usr/lib/spark/sbin/stop-thriftserver.sh'])
```

Reply:
@buggtb, danger!  this isn't going to work and the mighty @johnsca has confirmed the error of my ways.  Since the handler is a catch-all with 'config.changed', if something is set in addition to `spark_enable_thrift`, we'll skip the handler because of the @when_not -- basically forget my suggestion to `reconfigure_spark`.  We'll need to keep what you had to process config_changed, skipping the reinstall if our bigtop_version or spark_enable_thrift are set.

Still, I think the `start_thrift` method would nice to refactor to run @when(spark.started, spark_thrift_enabled) and drop the data_changed logic.

Reply:
"Still, I think the start_thrift method would be nice to refactor to run @when(spark.started, spark_thrift_enabled) and drop the data_changed logic." < can you react to a specific config change? I've not noticed that pattern in a charm before.

I see elsewhere in that charm bigtop.version.changed, but then it doesn't use that config option to look anything up.

Reply:
> can you react to a specific config change

yup, you're doing it right :)